frontend/packages/console-dynamic-plugin-sdk/src/utils/fetch/console-fetch-utils.ts

```diff
@@ -46,3 +47,27 @@ export const getConsoleRequestHeaders = (): ConsoleRequestHeaders => {
 
   return headers;
 };
+
+export const parseData = async (response) => {
+  const text = await response.text();
+  const isPlainText = response.headers.get('Content-Type') === 'text/plain';
+  if (!text) {
+    return isPlainText ? '' : {};
+  }
+  return isPlainText || !response.ok ? text : JSON.parse(text);
+};
+
+/**
+ * This function takes a response object, converts the response to JSON, and then returns the data, headers, and status.
+ * @param {Response} response - The response object to process.
+ * @return {Promise<{ data: R, headers: Headers, status: number }>} - A promise that resolves to an object containing the data, headers, and status of the response.
+ */
+export const getConsoleResponseDetails = async <R extends K8sResourceCommon>(
```

Comment:
Why does this function need to be async?

Reply:
I feel like the logic is simple enough that this doesn't need to be a public API for plugins.

Reply:
I'm a little confused as I thought we had the change the k8s* request APIs to get the response headers. Is that not the case?

Reply:
> I feel like the logic is simple enough that this doesn't need to be a public API for plugins.

@spadgett Sure, I can place the logic in console-shared package or else where. I placed it here since any one consuming `consoleFetch` entireHTTPResponse would likely need it.  So I placed it close to the `consoleFetch` method.

Reply:
> I'm a little confused as I thought we had the change the k8s* request APIs to get the response headers. Is that not the case?

@spadgett Not sure I understood the question here. However, yes, we extended `consoleFetchCommon` to return `headers` which would result in modifying the methods that invoke `consoleFetchCommon`  down the line.

Reply:
> Why does this function need to be async?

@spadgett Will double check if this is necessary.

Reply:
> Sure, I can place the logic in console-shared package or else where. I placed it here since any one consuming consoleFetch entireHTTPResponse would likely need it. So I placed it close to the consoleFetch method.

The important thing is to determine whether we've exposed this (and `parseData`) as a public API to plugins by moving it here. We should try to keep it internal I think.

Reply:
> Not sure I understood the question here. However, yes, we extended consoleFetchCommon to return headers which would result in modifying the methods that invoke consoleFetchCommon down the line.

Hmm, did I miss that change in this PR? Or was it updated in a different PR?

Reply:
> > Not sure I understood the question here. However, yes, we extended consoleFetchCommon to return headers which would result in modifying the methods that invoke consoleFetchCommon down the line.
> 
> Hmm, did I miss that change in this PR? Or was it updated in a different PR?

@spadgett See the PR - https://github.com/openshift/console/pull/13623 for extending support for returning response.header in  `consoleFetchCommon function`.

Reply:
> Why does this function need to be async?

@spadgett Because we are invoking an async `parseData`  function in line 69.